tests/test_bump_dictionary.py

```diff
@@ -0,0 +1,127 @@
+from typer.testing import CliRunner
+
+from bump_dictionary.cli import bump_dictionary
+
+
+@pytest.fixture(scope="session")
+def runner():
+    return CliRunner()
+
+
+@pytest.fixture(scope="session")
+def example_dictionaries_path():
+    return Path(__file__).absolute().parent / "data"
+
+
+@pytest.fixture(scope="function")
+def example_output_path(tmp_path):
+    return tmp_path / "updated_dictionary.json"
+
+
+@pytest.fixture(scope="session")
+def load_test_json():
+    def _read_file(file_path):
+        with open(file_path, "r") as f:
+            return json.load(f)
+
+    return _read_file
+
+
+def test_valid_legacy_dictionary_upgraded(
+    load_test_json, example_dictionaries_path, runner, example_output_path
+):
+    """Test that a data dictionary valid against the previous schema is upgraded correctly."""
+    target_latest_dict = load_test_json(
+        example_dictionaries_path / "latest_schema_dictionary.json"
+    )
+
+    result = runner.invoke(
+        bump_dictionary,
+        [
+            str(example_dictionaries_path / "legacy_schema_dictionary.json"),
+            str(example_output_path),
+        ],
+    )
+
+    output = load_test_json(example_output_path)
+
+    assert result.exit_code == 0
+    assert output == target_latest_dict
+
+
+def test_valid_legacy_dictionary_with_transformation_upgraded(
+    load_test_json,
+    example_dictionaries_path,
+    runner,
+    example_output_path,
+    caplog,
+):
+    """
+    Test that a valid legacy data dictionary containing 'Transformation' for continuous columns
+    is upgraded correctly, with 'Transformation' renamed to 'Format'.
+    """
+    target_latest_dict = load_test_json(
+        example_dictionaries_path / "latest_schema_dictionary.json"
+    )
+
+    result = runner.invoke(
+        bump_dictionary,
+        [
+            str(
+                example_dictionaries_path
+                / "legacy_schema_dictionary_with_transformation.json"
+            ),
+            str(example_output_path),
+        ],
+    )
+
+    output = load_test_json(example_output_path)
+
+    assert result.exit_code == 0
+    assert "Renaming 'Transformation' to 'Format'" in caplog.text
+    assert output == target_latest_dict
+
+
+def test_valid_latest_dictionary_not_upgraded(
+    example_dictionaries_path, runner, caplog
+):
+    """
+    Test that a data dictionary valid against the latest schema is not upgraded,
+    with an informative error.
+    """
+    result = runner.invoke(
+        bump_dictionary,
+        [
+            str(example_dictionaries_path / "latest_schema_dictionary.json"),
+            str(example_output_path),
+        ],
+    )
+
+    assert result.exit_code != 0
+    assert len(caplog.records) == 1
+    assert "already up-to-date" in caplog.text
+
+
+def test_invalid_dictionary_not_upgraded(
+    example_dictionaries_path, runner, caplog
+):
+    """
+    Test that a data dictionary which is not valid against the previous schema is not upgraded,
```

Comment:
```suggestion
    Test that a data dictionary which is not valid against the legacy schema is not upgraded,
```
Let's stick to same names - whatever they are